www/js/app.js

```diff
@@ -2303,6 +2318,112 @@ function articleLoadedSW (iframeArticleContent) {
+    if (appstate.wikimediaZimLoaded && params.showPopoverPreviews && 'matches' in Element.prototype) {
+        const iframeDoc = win.document;
+        if (iframeDoc) {
+            // Attach the popover CSS to the current article document
+            uiUtil.attachKiwixPopoverCss(iframeDoc);
+            // Add event listeners to iframe window to check for links
+            win.addEventListener('mouseover', handlePopoverEvents, true);
+            win.addEventListener('focus', handlePopoverEvents, true);
+            // Conditionally add event listeners to support touch events with fallback to pointer events
+            if (window.navigator.maxTouchPoints > 0) {
+                win.addEventListener('touchstart', handlePopoverEvents, true);
+            } else {
+                win.addEventListener('pointerdown', handlePopoverEvents, true);
+            }
+        }
+    }
+}
+
+/**
+ * Event handler for attaching preview popovers
+ * @param {Event} event The event produced by the mouseover or focus action
+ */
+function handlePopoverEvents (event) {
+    // Check if the hovered or focused element or its parent is a link (but throttle its activation)
+    if (window.popoverThrottle) return;
+    window.popoverThrottle = true;
+    setTimeout(function () {
+        let a = event.target;
+        const iframeDoc = a.ownerDocument;
+        if (iframeDoc) {
+            const iframeWindow = iframeDoc.defaultView;
+            while (a && a !== iframeWindow && a.nodeName !== 'A') {
+                a = a.parentNode;
+            }
+            // If a link was hovered or focused, process it
+            if (a && a.nodeName === 'A') {
+                // console.debug(event.type, event.target, a);
+                const suppressContextMenuHandler = function (e) {
+                    e.preventDefault();
+                    e.stopPropagation();
+                };
+                // Prevent context menu on this anchor element
+                a.addEventListener('contextmenu', suppressContextMenuHandler, true);
+                if (/touchstart|pointerdown/.test(event.type)) {
+                    a.touched = true; // Used to prevent dismissal of popver on mouseout if initiated by touch
+                }
+                if (a.style.userSelect === undefined) {
+                    // This prevents selection of the text in a touched link in Safari for iOS and Edge Legacy / UWP
+                    a.style.webkitUserSelect = 'none';
+                    a.style.msUserSelect = 'none';
+                }
+                // Check if a popover div is currently being hovered
+                const divs = iframeDoc.getElementsByClassName('kiwixtooltip');
+                let divIsHovered = false;
+                Array.prototype.slice.call(divs).forEach(function (div) {
+                    if (div.matches(':hover')) divIsHovered = true;
```

Comment:
Not sure of your polyfill situation, can you use Array.some?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

So it would be `const divIsHoevered = divs.some(() => div.matches(':hover'))` in modern, say, Typescript

Reply:
As mentioned above, we have `Array.from` polyfilled. It would be interesting to see if core-js picks up Array.some -- will try, but I suspect I'll be limited to Array.from.

Reply:
Turns out Array.some() has wide support without polyfills, so I've incorporated it, thanks!